### ISSUE TITLE
The video url player will crash STARK

### ISSUE THREAD

**Issue by TheBorker** [GrimReaperFloof/STARK#3]
It will crash if u try to use the url player

**Comment by fuck-shithub** [comment]
Yeah I'm no longer maintaining this program. If you want to get your hands dirty, you can try updating YoutubeExplode yourself and see if that works, as that is likely the faulty library. I'm no longer interested in updating this program, as I've stopped using Windows (the only OS this program supports), and I have stopped playing Source games because I also stopped using Steam.

I began with this project as a way to get my hands dirty myself. There was a program called TF2Speech at the time. It was a paid program, and it essentially does the same things as this project, but with some other features like allowing players to use other TTS voices dynamically. I didn't want to pay any money, so I put hours of hard work and free time into creating my own bot, based on the works of axynos which had already done a lot of work on it. But things don't last forever. TF2Speech's website appears to be dead. [Archive available here.](https://web.archive.org/web/20180829133741/https://tf2speech.com/). Sadly, due to the fact it is a paid program, you cannot download this anymore. If the developer stops providing the service, it disappears. That's why proprietary projects like TF2Speech were doomed to die and completely disappear from the internet. Good luck finding a copy. At least my project is still up, source code available, but it is just no longer maintained.

I found some other projects similar to this one that might be interesting:
https://github.com/Joey-Einerhand/TF2ChatToSpeech - A Windows TTS program, probably has less features than STARK.
https://steamcommunity.com/sharedfiles/filedetails/?id=1747696293 - A very simple TTS script for Linux. I'm quite surprised this even exists.
https://github.com/fuck-shithub/SourceDJ - I wrote this program as an attempt to make a cross-platform TTS bot similar to STARK. It can do TTS and YouTube playback, but is also unmaintained. It has a very simple architecture that should be easy to hack on, though.
